### ISSUE TITLE
Sorts K and KItem are incompatible

### ISSUE THREAD

**Issue by norbeyandresg** [kframework/evm-semantics#232]
Trying to prove a simple spec file I got this error. Can someone help me?
`
error "line 21 column 135: Sorts K and KItem are incompatible"`

Here is the spec file.

```
requires "pow-verification.k"

module POW-OF-TWO-SPEC
  imports ETHEREUM-SIMULATION
  imports VERIFICATION

  //powOfTwo
  rule <k> #execute ... </k>
       <mode> NORMAL </mode>
       <schedule> DEFAULT </schedule>
       <callStack> .List </callStack>
       <memoryUsed> 0 </memoryUsed>
       <localMem> .Map </localMem>
       <previousGas>  _ => _ </previousGas>
       <program> powOfTwo(N) </program>

       <pc>        0 => 56                           </pc>
       <wordStack> WS => 0 : 2 ^Int N : WS           </wordStack>
       <gas>       G  => _  </gas>

       requires N >=Int 0
       andBool  N <=Int 340282366920938463463374607431768211455
       andBool  #sizeWordStack(WS) <Int 1021

  rule <k> #execute ... </k>
       <mode> NORMAL </mode>
       <schedule> DEFAULT </schedule>
       <callStack> .List </callStack>
       <memoryUsed> 0 </memoryUsed>
       <localMem> .Map </localMem>
       <previousGas>  _ => _ </previousGas>
       <program> powOfTwo(N) </program>

       <pc>        35 => 56                           </pc>
       <gas>       G  => _   </gas>
       <wordStack> I : S         : WS
                => 0 : S +Int S  : WS                 </wordStack>

       requires I >=Int 0
       andBool  S >=Int 0
       andBool  S +Int S ==Int 2 ^Int I
       andBool  S +Int S <=Int pow256
       andBool  #sizeWordStack(WS) <Int 1021
endmodule
```

**Comment by ehildenb** [comment]
Is it dumping the query to Z3 that is being made? Unfortunately running `./kevm prove` does not allow dumping the Z3 queries right now because it doesn't pass the option through.

This is fixed in https://github.com/kframework/evm-semantics/pull/233. If you checkout that branch then you should be able to add `--debug` to the call to `./kevm prove` and it will dump more information.

**Comment by norbeyandresg** [comment]
Hi, this is what I get using `./kevm prove`

```
[WARNING] Running as root is not recommended
[Error] Critical: Z3 crashed on input query:
(declare-sort WordStack)
(declare-sort K)
(declare-sort KItem)
(declare-fun sizeWordStackAux (WordStack Int) Int)
(declare-fun sizeWordStack (WordStack) Int)
(declare-fun smt_keccak (WordStack) Int)
(declare-fun asByteStack (Int WordStack) WordStack)
(declare-fun chop (Int) Int)
(declare-fun smt_nthbyteof (Int Int Int) Int)
(declare-fun asWord (WordStack) Int)
(assert (forall ((|_998| Int)) (= (chop |_998|) (mod |_998|
115792089237316195423570985008687907853269984665640564039457584007913129639936))))
(assert (forall ((|_657| WordStack)(|_658| Int)) (= (>= (sizeWordStackAux
|_657| |_658|) 0) true)))
(declare-fun |R_BAL| () Int)
(declare-fun |R_VALUE| () Int)
(declare-fun |_1229| () Int)
(declare-fun |R_CD| () Int)
(declare-fun |_1230| () Int)
(declare-fun |_1233| () WordStack)
(declare-fun |_1232| () Int)
(declare-fun |_1883| () Int)
(declare-fun |_1884| () K)
(declare-fun |_1887| () K)
(declare-fun |_1885| () KItem)
(declare-fun |_1886| () K)
(assert (and (= (< |_1229| 0) true) (= (>= |_1230| 0) true) (= (+ |_1232|
|_1232|) _1883) (= _1884 _1885) (= (or (> |R_VALUE| |R_BAL|) (>= |R_CD| 1024))
false) (= (<= (+ |_1232| |_1232|)
115792089237316195423570985008687907853269984665640564039457584007913129639936)
true) (= (>= |_1232| 0) true) (= _1886 _1887) (= (< (sizeWordStackAux |_1233|
0) 1021) true)))
result:
(error "line 25 column 104: Sorts K and KItem are incompatible")
[Warning] Critical: missing SMTLib translation for #KSequence (missing SMTLib
translation for #KSequence)

```


and this what I use `./kevm debug`

```
== debugging: tests/proofs/resources/evm-specs/pow-of-two-spec.k
[WARNING] Running as root is not recommended
[Error] Critical: Parser returned a non-zero exit code: 113
Stdout:

Stderr:
[Error] Inner Parser: Scanner error: unexpected character sequence '-'.
        Source(/evm-semantics/./tests/proofs/resources/evm-specs/pow-of-two-spec.k)
        Location(3,11,3,12)
```

**Comment by norbeyandresg** [comment]
@ehildenb can you give me some help with this?

**Comment by dteiml** [comment]
@norbeyandresg Seems like the error is thrown at the `-` character in your module title. Maybe try removing the dashes from module name?

**Comment by ehildenb** [comment]
I believe this has been fixed (duplicate of #192 ?). Re-open if not the case.
